Maybe not useful exception on socket reset in HttpSyncer

Could probably eat this exception

###### Reproduction Steps

1. Nothing really in the logcat 

> 05-13 12:24:26.330 E/AnkiDroid(30583): Connection/ doInBackgroundSync error

Just the exception

###### Expected Result

should not send an exception report unless it is actionable, but more likely the user needs a toast or something, vs us getting an exception report

###### Actual Result

```
java.lang.RuntimeException: java.net.SocketException: Connection reset
at com.ichi2.libanki.sync.HttpSyncer.req(HttpSyncer.java:52)
at com.ichi2.libanki.sync.HttpSyncer.req(HttpSyncer.java:2)
at com.ichi2.libanki.sync.RemoteServer.meta(RemoteServer.java:9)
at com.ichi2.libanki.sync.Syncer.sync(Syncer.java:4)
at com.ichi2.async.Connection.doInBackgroundSync(Connection.java:16)
at com.ichi2.async.Connection.doOneInBackground(Connection.java:2)
at com.ichi2.async.Connection.doInBackground(Connection.java:4)
at com.ichi2.async.Connection.doInBackground(Connection.java:1)
at android.os.AsyncTask$3.call(AsyncTask.java:394)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:923)
Caused by: java.net.SocketException: Connection reset
at java.net.SocketInputStream.read(SocketInputStream.java:215)
at java.net.SocketInputStream.read(SocketInputStream.java:144)
at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.readFromSocket(ConscryptEngineSocket.java:945)
at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.processDataFromSocket(ConscryptEngineSocket.java:909)
at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.access$100(ConscryptEngineSocket.java:731)
at com.android.org.conscrypt.ConscryptEngineSocket.doHandshake(ConscryptEngineSocket.java:241)
at com.android.org.conscrypt.ConscryptEngineSocket.startHandshake(ConscryptEngineSocket.java:220)
at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.kt:9)
at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.kt:9)
at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:17)
at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:39)
at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:1)
at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:7)
at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:8)
at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:2)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:12)
at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:27)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:12)
at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:22)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:12)
at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:7)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:12)
at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:16)
at okhttp3.internal.connection.RealCall.execute(RealCall.kt:5)
at com.ichi2.libanki.sync.HttpSyncer.req(HttpSyncer.java:39)
```


###### Debug info

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/28c44e01-e62b-4aa2-a73e-64b1292867f5


###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

--- a/AnkiDroid/src/main/java/com/ichi2/async/Connection.java
+++ b/AnkiDroid/src/main/java/com/ichi2/async/Connection.java
@@ -290,6 +290,8 @@ public class Connection extends BaseAsyncTask<Connection.Payload, Object, Connec
                 msg.contains("Failed to connect") ||
                 msg.contains("InterruptedIOException") ||
                 msg.contains("stream was reset") ||
+                msg.contains("Connection reset") ||
+                msg.contains("Broken pipe") ||
                 msg.contains("ConnectionShutdownException") ||
                 msg.contains("CLEARTEXT communication") ||
                 msg.contains("TimeoutException");

